FiveOhFirstDataCore.Pages/Data/PublicTrooperProfile.razor

```diff
@@ -110,15 +111,15 @@
                                     {
                                         @if((LoggedIn?.Id ?? 0) == Trooper.Id || LoadedAdditional)
                                         {
-                                                <tr draggable="true" @ondrag="()=>OnDrag(desc)" @ondrop="()=>OnDrop(desc)">
+                                                <tr draggable="true" @ondrag="()=>OnDrag(desc)" @ondrop="async ()=>OnDrop(desc)">
                                                     <td>@desc.Content</td>
                                                     <td>@desc.CreatedOn.ToString("d")</td>
-                                                    <td><button @onclick="()=>DeleteDescription(desc)" class="btn btn-link"><span class="oi oi-trash"></span></button></td>
+                                                    <td><button @onclick="async ()=>DeleteDescription(desc)" class="btn btn-link"><span class="oi oi-trash"></span></button></td>
                                                 </tr>
```

Comment:
This section will have a logic error because you are using the iterator and not a local variable in your method calls. It will always use the last description in list no mater what object you move.

Do something such as `var d = desc;` directly after the foreach statement (exclude the @ sign, you dont need it if its code following code) and use the variable `d` in the method params not `desc`.